specifications/datastore-environment-manager.org

```diff
@@ -0,0 +1,123 @@
+
+- When migrating a database, each migration will be logged in the datastore for triaging purposes.
+
+- Any datastore without this table will be considered invalid.
+
+*** BuildEnvironments
+
+The datastore will have a =BuildEnvironments= table with an array of =build-environment= records:
+
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+| field              | type   | description                                                                                  | example                                             |
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+| provider           | string | Name of build provider, e.g. ["host", "lxd", "multipass", "launchapad"]                      | "multipass"                                         |
+| timestamp_created  | string | Creation timestamp of build environment instance using python's =time.asctime()=.            | "Fri Aug 21 15:56:20 2020"                          |
+| timestamp_accessed | string | Accessed (last use) timestamp of build environment instance using python's =time.asctime()=. | "Fri Aug 21 15:56:20 2020"                          |
+| snapcraft_version  | string | Version of snapcraft that created this environment.                                          | "4.0.1"                                             |
+| project_name       | string | Name of project (as derived from snapcraft.yaml).                                            | "my-snap-name"                                      |
+| project_path       | string | Path to project.                                                                             | "/home/user/git/my-snap-name"                       |
+| project_id         | string | Unique identifier for project.                                                               | "<project-id>" as defined in environment lifecycle. |
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+
+- Whenever an environment is created or updated, the record is created/updated.  When cleaning the environment, the =build-environment= record is purged.
+
+*** LXD
+
+The datastore will have a =LXD= table with an array of =lxd-environment= records:
+
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+| field              | type   | description                                                                                  | example                                             |
```

Comment:
the remote used might be a useful field here too (for remote-build potential)

Reply:
the remote was baked into the image name, or at least how I expected to use `image name`  maybe `image_source` would be most appropriate?